src/components/scenes/WalletRestoreScene.tsx

```diff
@@ -0,0 +1,185 @@
+import { EdgeWalletInfoFull } from 'edge-core-js'
+import * as React from 'react'
+import { FlatList, View } from 'react-native'
+
+import { showRestoreWalletsModal } from '../../actions/SettingsActions'
+import { useHandler } from '../../hooks/useHandler'
+import { useWatch } from '../../hooks/useWatch'
+import { toLocaleDate } from '../../locales/intl'
+import { lstrings } from '../../locales/strings'
+import { useDispatch, useSelector } from '../../types/reactRedux'
+import { EdgeSceneProps } from '../../types/routerTypes'
+import { FlatListItem } from '../../types/types'
+import { normalizeForSearch } from '../../util/utils'
+import { SCENE_BUTTONS_MARGIN_REM, SceneButtons } from '../buttons/SceneButtons'
+import { WalletRestoreCard } from '../cards/WalletRestoreCard'
+import { EdgeAnim, fadeInLeft, fadeInRight, fadeOut } from '../common/EdgeAnim'
+import { InsetStyle, SceneWrapper } from '../common/SceneWrapper'
+import { SearchIconAnimated } from '../icons/ThemedIcons'
+import { cacheStyles, Theme, useTheme } from '../services/ThemeContext'
+import { SceneHeaderUi4 } from '../themed/SceneHeaderUi4'
+import { SimpleTextInput } from '../themed/SimpleTextInput'
+import { WalletListSectionHeader } from '../themed/WalletListSectionHeader'
+
+interface Props extends EdgeSceneProps<'walletRestore'> {}
+
+export function WalletRestoreScene(props: Props) {
+  const { navigation } = props
+  const dispatch = useDispatch()
+  const theme = useTheme()
+  const styles = getStyles(theme)
+
+  const account = useSelector(state => state.core.account)
+  const allKeys = useWatch(account, 'allKeys')
+  const restoreWalletInfos = allKeys.filter(key => key.archived || key.deleted).reverse()
```

Comment:
The order of the keys is semi-random, so `reverse` isn't as useful as it seems. We should really sort the keys by date.

```ts
allKeys.filter(...).sort((a, b) => {
  const aDate = a.created?.valueOf() ?? 0
  const bDate = b.created?.valueOf() ?? 0
  return aDate - bDate
})
```